link/executable.go

```diff
@@ -0,0 +1,78 @@
+package link
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+var ErrSymbolNotFound error
+
+type executable struct {
```

Comment:
Unexported types in the API aren't ergonomic for users. `type Executable` here and `OpenExecutable` or similar below?